code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -38,19 +49,15 @@
 	var/obj/item/slimepotion/slime/current_potion
 	var/obj/machinery/monkey_recycler/connected_recycler
 
-	icon_screen = "slime_comp"
-	icon_keyboard = "rd_key"
-
 /obj/machinery/computer/camera_advanced/xenobio/Initialize(mapload)
-	..()
-	return INITIALIZE_HINT_LATELOAD
+	. = ..()
+	locate_recycler()
 
-/obj/machinery/computer/camera_advanced/xenobio/LateInitialize()
-	..()
-	for(var/obj/machinery/monkey_recycler/recycler in GLOB.monkey_recyclers)
-		if(get_area(recycler.loc) == get_area(loc))
-			connected_recycler = recycler
-			connected_recycler.connected += src
+/obj/machinery/computer/camera_advanced/xenobio/proc/locate_recycler()
+	for(var/obj/machinery/monkey_recycler/recycler in get_area(loc))
```

Comment:
Looping over an area is super expensive IIRC. Performance wise it might be better to keep the original code.
Worth testing though.
Alternatively instead of having a loop over the recyclers. You can have the recycler add itself to an assoc list with the area as key. Though realistically you'd only really have one or two monkey recyclers active max